src/mixins.coffee

```diff
@@ -64,3 +64,75 @@ Ember.Widgets.BodyEventListener = Ember.Mixin.create
   _removeDocumentHandlers: ->
     $(@get('bodyElementSelector')).off "click", @_clickHandler
     @_clickHandler = null
+
+Ember.Widgets.TabbableModal = Ember.Mixin.create
+  currentFocus: null
+
+  _focusTabbable: ->
+     # Set focus to the first match:
+     # 1. The saved focused element
+     # 2. First element inside the dialog matching [autofocus]
+     # 3. Tabbable element inside the content element
+     # 4. The close button (has class "close")
+    hasFocus = [@get 'currentFocus']
+    if hasFocus[0] == null or not $.contains(@$()[0], hasFocus[0])
+      hasFocus = @$( "[autofocus]" )
+    if hasFocus.length == 0
+      hasFocus = @$( ":tabbable" )
+    if hasFocus.length > 0
+      focusElement = hasFocus[0]
+      if focusElement.className.indexOf("close") > -1
+        # if we have more than two tabbable objects, we do not want to tab
+        # to close button
+        # while if we do not have any choice, the close button is chosen
+        if hasFocus.length > 1
+          focusElement = hasFocus[1]
+      focusElement.focus()
+      @set 'currentFocus', focusElement
+
+
+  _checkContainingElement: (containers, element) ->
+    for containerItem in containers
+      if containerItem is element or $.contains(containerItem, element)
+        return yes
+    return no
+
+  click: (event) ->
+    isActive = $.contains(@$()[0], event.target)
+    if not isActive
+      @hide() unless @get('enforceModality')
+      @_focusTabbable()
+    if not @_checkContainingElement(@$(':focusable'), event.target)
+      @_focusTabbable()
+
+  # capture the TAB key and make a cycle tab loop among the tabbable elements
+  # inside the modal. Remove the close button from the loop
+  keyDown: (event) ->
+    return if event.isDefaultPrevented()
+
+    if event.keyCode == @KEY_CODES.ESCAPE and @get 'escToCancel'
+      @doCancelation()
+      event.preventDefault()
+    else if event.keyCode == @KEY_CODES.TAB
+      # tabbable objects list without close button
+      tabbableObjects = @$(":tabbable").not('.close')
+
+      _currentFocus = $(document.activeElement)?[0]
+
+      _index =tabbableObjects.index _currentFocus
+      if _index == -1
+        @_focusTabbable()
+      if (tabbableObjects.length > 0)
+        first = tabbableObjects[0]
+        last = tabbableObjects[tabbableObjects.length - 1]
+        # check the two ends of the array to make it the tab loop
+        if (event.target == last and not event.shiftKey)
+          first.focus()
+          @set 'currentFocus', first
+          event.preventDefault()
+        else if (event.target == first and event.shiftKey)
+          @set 'currentFocus', last
+          last.focus()
+          event.preventDefault()
+        else
+          @set 'currentFocus', tabbableObjects[_index+1]
```

Comment:
also, can you explain how this actually works? I don't see you calling `.focus()` anywhere.